### PR TITLE
Add unload_textual_inversion method

### LINKED ISSUE #6013: Add `unload_textual_inversion()` method
**Is your feature request related to a problem? Please describe.**
Currently you can add textual inversion with `.load_textual_inversion()`. But once the tokenizer and text encoder have the new embeddings, it's complex to remove the embeddings and get it back to the original state.

**Describe the solution you'd like.**
An `unload_textual_inversion()` to remove alien/foreign tokens and get the text encoder back to the original state (maybe there could also be a way to pass specific tokens to be removed). 

**Describe alternatives you've considered.**
As @patrickvonplaten described internally, it is possible to remove tokens currently by doing something like:
```py
token_id = pipe.tokenizer.convert_token_to_id("<token>")
del pipe.tokenizer._added_tokens_decoder[token_id]
pipe.tokenizer._update_trie()
```
and
```py
text_embeddings = pipe.text_encoder.get_input_embeddings().weight
text_embeddings = text_embedding[:len(pipe.tokenizer)]
pipe.text_encoder.set_input_embeddings(text_embeddings)
```
and it would probably be easier with a specific `remove_token` method from Transformers (https://github.com/huggingface/transformers/issues/15032, https://github.com/huggingface/transformers/issues/4827). 
However, imo it does not eliminate the usefulness of having a diffusers specific method that leverages the ways of removing tokens in order to provide a simple API for folks to remove tokens. 

**Additional context.**
Hot-swapping (keeping a base model warm and swapping the LoRAs on top) requires fusing/unfusing LoRAs, but with pivotal tuning this becomes more complex to reset the text encoder, tokenizer

### PR DESCRIPTION
**Pull request by fabiorigano**
# What does this PR do?

Fixes #6013 

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@apolinario @sayakpaul

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@yiyixuxu could you try to give this a review?

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:497]: Shouldn't we only try to remove the specific `tokens` if it is not `None`?
i.e. we would try to find `token_id` of the `tokens` user passed (e.g.`for token in tokenizer(token)["input_ids"] ...`) instead of looping through `added_token_decoder.items()`

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:521]: can you explain why do we need this block?

**Review by yiyixuxu — COMMENTED** [review]
thanks for working on this! I left some questions :)

File: src/diffusers/loaders/textual_inversion.py
Comment (fabiorigano) [reply src/diffusers/loaders/textual_inversion.py:497]: yes, we can retrieve the token_id with tokenizer(token)["input_ids"][1]. 
I was looping through `added_token_decoder.items()` to avoid removing non-Textual Inversion tokens and special tokens

Comment (yiyixuxu) [reply src/diffusers/loaders/textual_inversion.py:497]:
```suggestion
            for added_token_id, added_token in tokenizer.added_tokens_decoder.items():
```

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:500]:
```suggestion
                        token_ids.append(added_token_id)
```

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:507]:
```suggestion
            for added_token_id, added_token in tokenizer.added_tokens_decoder.items():
```

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:509]:
```suggestion
                    token_ids.append(added_token_id)
```

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [inline src/diffusers/loaders/textual_inversion.py:512]:
```suggestion
                    last_special_token_id = added_token_id
```

File: src/diffusers/loaders/textual_inversion.py
Comment (yiyixuxu) [reply src/diffusers/loaders/textual_inversion.py:521]: thanks for explaining this! @fabiorigano 

I'm not very familiar with the use case 
cc @apolinario ad @linoytsaban here can you take a look to see if we need to reorder the remaining added tokens after we remove some?

**Review by yiyixuxu — APPROVED** [review]
looks good to me! thanks for working on this:)
let's wait for a final review from @apolinario

File: src/diffusers/loaders/textual_inversion.py
Comment (linoytsaban) [reply src/diffusers/loaders/textual_inversion.py:521]: Thanks @fabiorigano! I looked a bit but I'm actually not quite sure why it's necessary to reorder 🤔

**Review by linoytsaban — COMMENTED** [review]
Thanks a lot for this PR @fabiorigano! LGTM :)

File: src/diffusers/loaders/textual_inversion.py
Comment (fabiorigano) [reply src/diffusers/loaders/textual_inversion.py:521]: hi @linoytsaban, the reordering block is useful to have the same indeces as the text embeddings in the encoder, so multiple `unload_textual_inversion(<token-x>)` calls will remove the correct text embeddings. If the reordering is not done, when re-executing `unload_textual_inversion(<token-y>)` the last for loop may fail, because it may remove a different text embedding than expected.

Comment (linoytsaban) [reply src/diffusers/loaders/textual_inversion.py:521]: gotcha! that makes total sense, thanks for explaining! 🤗

**Review by linoytsaban — APPROVED** [review]

**Comment by yiyixuxu** [comment]
@fabiorigano @linoytsaban thank you both!

great work as always @fabiorigano :)

**Comment by fabiorigano** [comment]
@yiyixuxu @linoytsaban thank you for your reviews! :)

**Pull request merged**
